tracer/src/Datadog.Trace/Ci/Agent/Payloads/CICodeCoveragePayload.cs

```diff
@@ -0,0 +1,83 @@
+// <copyright file="CICodeCoveragePayload.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Text;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Agent.Transports;
+using Datadog.Trace.Ci.Coverage.Models;
+using Datadog.Trace.Vendors.MessagePack;
+
+namespace Datadog.Trace.Ci.Agent.Payloads
+{
+    internal class CICodeCoveragePayload : MultipartPayload
+    {
+        public CICodeCoveragePayload(int maxItemsPerPayload = DefaultMaxItemsPerPayload, int maxBytesPerPayload = DefaultMaxBytesPerPayload, IFormatterResolver formatterResolver = null)
+            : base(maxItemsPerPayload, maxBytesPerPayload, formatterResolver)
+        {
+            var agentlessUrl = CIVisibility.Settings.AgentlessUrl;
+            if (!string.IsNullOrWhiteSpace(agentlessUrl))
+            {
+                var builder = new UriBuilder(agentlessUrl);
+                builder.Path = "api/v2/citestcov";
+                Url = builder.Uri;
+            }
+            else
+            {
+                var builder = new UriBuilder("https://datadog.host.com/api/v2/citestcov");
+                builder.Host = "event-platform-intake." + CIVisibility.Settings.Site;
```

Comment:
I find this a bit strange to create with an invalid host and then replace it - could we use the other constructor instead?
```suggestion
                var builder = new UriBuilder(
                    scheme: "https",
                    host: "event-platform-intake." + CIVisibility.Settings.Site,
                    port: 443,
                    pathValue: "api/v2/citestcov);
```

Reply:
done in [08095f3](https://github.com/DataDog/dd-trace-dotnet/pull/2983/commits/08095f3ce582bf4337ed4e5e4187a9c50e531817)